File: software_challenge_sdk/src/server_api/sc/protocol/requests/AuthenticateRequest.java
package sc.protocol.requests;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;
import sc.protocol.responses.ProtocolMessage;

/**
 * Send by Client to authenticate as administrator. Is not answered if successful.
 */

@XStreamAlias("authenticate")
public class AuthenticateRequest extends ProtocolMessage implements ILobbyRequest {
  @XStreamAsAttribute
  private String passphrase;

  public AuthenticateRequest(String passphrase) {
    this.passphrase = passphrase;
  }

  public String getPassword() {
    return this.passphrase;
  }
}

File: socha-sdk/src/server-api/sc/protocol/requests/AuthenticateRequest.kt
package sc.protocol.requests

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute

/** Sent by Client to authenticate as administrator. Is not answered if successful. */
@XStreamAlias("authenticate")
data class AuthenticateRequest(
        @XStreamAsAttribute
        val password: String
): ILobbyRequest
